tests/unittests/test_upgrade.py

```diff
@@ -36,6 +41,17 @@ def previous_obj_pkl(self, request):
         """
         return pkl_load(str(request.param))
 
+    def test_pkl_load_defines_all_init_side_effect_attributes(
+        self, previous_obj_pkl
+    ):
+        """Any attrs as side-effects of __init__ exist in unpickled obj."""
+        ds_class = DSNAME_TO_CLASS[previous_obj_pkl.dsname]
+        sys_cfg = previous_obj_pkl.sys_cfg
+        distro = previous_obj_pkl.distro
```

Comment:
Can we instanciate and compare distro too?

Reply:
Since we pickle distros and could just as easily be bitten with distro as well could we do a similar `__dict__` comparison of a pickled object and a new object?

Reply:
@holmanb +1 on this too. I think we should capture this as a work item and add it as separate  PR. (like some of the other datasource types beyond Azure and NoCloud